openedx_lti_tool_plugin/deep_linking/tests/test_views.py

```diff
@@ -0,0 +1,268 @@
+    def test_with_cached_launch_id(
+        self,
+        http_response_mock: MagicMock,
+        validate_message_mock: MagicMock,
+        get_message_from_cache_mock: MagicMock,
+    ):
+        """Test with the launch_id of a cached DjangoMessageLaunch object."""
+        response = self.view_class.as_view()(self.request, self.launch_id)
+
+        get_message_from_cache_mock.assert_called_once_with(self.request, self.launch_id)
+        validate_message_mock.assert_called_once_with(get_message_from_cache_mock())
+        http_response_mock.assert_called_once_with()
+        self.assertEqual(response, http_response_mock())
+
+    @patch.object(DeepLinkingFormView, 'http_response_error')
+    def test_with_lti_exception(
+        self,
+        http_response_error_mock: MagicMock,
+        validate_message_mock: MagicMock,
+        get_message_from_cache_mock: MagicMock,
+    ):
+        """Test with LtiException."""
+        exception = LtiException('Error message')
+        get_message_from_cache_mock.side_effect = exception
+
+        response = self.view_class.as_view()(self.request, self.launch_id)
+
+        get_message_from_cache_mock.assert_called_once_with(self.request, self.launch_id)
+        validate_message_mock.assert_not_called()
+        http_response_error_mock.assert_called_once_with(exception)
+        self.assertEqual(response, http_response_error_mock())
+
+    @patch.object(DeepLinkingFormView, 'http_response_error')
+    def test_with_deep_linking_exception(
+        self,
+        http_response_error_mock: MagicMock,
+        validate_message_mock: MagicMock,
+        get_message_from_cache_mock: MagicMock,
+    ):
+        """Test with DeepLinkingException."""
+        exception = DeepLinkingException('Error message')
+        validate_message_mock.side_effect = exception
+
+        response = self.view_class.as_view()(self.request, self.launch_id)
+
+        get_message_from_cache_mock.assert_called_once_with(self.request, self.launch_id)
+        validate_message_mock.assert_called_once_with(get_message_from_cache_mock())
+        http_response_error_mock.assert_called_once_with(exception)
+        self.assertEqual(response, http_response_error_mock())
+
+
+@patch.object(DeepLinkingFormView, 'tool_config', new_callable=PropertyMock)
+@patch.object(DeepLinkingFormView, 'tool_storage', new_callable=PropertyMock)
+@patch(f'{MODULE_PATH}.DjangoMessageLaunch')
+class TestDeepLinkingFormViewGetMessageFromCache(TestCase):
+    """Test DeepLinkingFormView get_message_from_cache method."""
+
+    def test_with_cached_launch_id(
+        self,
+        message_launch_mock: MagicMock,
+        tool_storage_mock: MagicMock,
+        tool_conf_mock: MagicMock,
+    ):
+        """Test with the launch_id of a cached DjangoMessageLaunch object."""
+        request = MagicMock()
+        launch_id = uuid4()
+
+        result = DeepLinkingFormView().get_message_from_cache(request, launch_id)
+
```

Comment:
@kuipumu I think you can call the method inside the assertEqual and even validate the calls.
```suggestion
self.assertEqual(
    DeepLinkingFormView().get_message_from_cache(request, launch_id), 
    message_launch_mock.from_cache(),
)
```